src/main/java/org/ethereum/beacon/discovery/storage/NodeTableImpl.java

```diff
@@ -163,35 +163,35 @@ private void loadNextBatch() {
           // Bounds are reached from both top and bottom
           break;
         }
-        if (upNodesOptional.isPresent()) {
-          NodeIndex upNodes = upNodesOptional.get();
-          for (Bytes currentNodeId : upNodes.getEntries()) {
-            if (logLimit == 0) {
-              res.add(getNode(currentNodeId).orElseThrow());
-            } else if (Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
-              limitReached = true;
-              break;
-            } else {
-              res.add(getNode(currentNodeId).orElseThrow());
-            }
-          }
-        }
-        if (downNodesOptional.isPresent()) {
-          NodeIndex downNodes = downNodesOptional.get();
-          List<Bytes> entries = downNodes.getEntries();
-          // XXX: iterate in reverse order to reach logDistance limit from the right side
-          for (int i = entries.size() - 1; i >= 0; i--) {
-            Bytes currentNodeId = entries.get(i);
-            if (logLimit == 0) {
-              res.add(getNode(currentNodeId).orElseThrow());
-            } else if (Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
-              limitReached = true;
-              break;
-            } else {
-              res.add(getNode(currentNodeId).orElseThrow());
-            }
-          }
-        }
+        upNodesOptional.ifPresent(
+            upNodes -> {
+              for (Bytes currentNodeId : upNodes.getEntries()) {
+                if (logLimit == 0) {
+                  getNode(currentNodeId).ifPresent(res::add);
+                } else if (Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
+                  limitReached = true;
+                  break;
+                } else {
+                  getNode(currentNodeId).ifPresent(res::add);
+                }
```

Comment:
(nit) Can this be simplified?:
```suggestion
                if (logLimit != 0 && Functions.logDistance(currentNodeId, nodeId) >= logLimit) {
                  limitReached = true;
                  break;
                } else {
                  getNode(currentNodeId).ifPresent(res::add);
                }
```

Reply:
Yep that works.